ngi_reports/common/ign_sample_report.py

```diff
@@ -274,11 +274,17 @@ def parse_picard_metrics(self):
     def create_aggregate_statistics (self):
 
         def create_header (samples):
-            return samples.values()[0].keys()
+            # take the union of all keys for the header row
+            return sorted(
+                list(
+                    set(
+                        [key \
+                        for sample in samples.values() \
+                        for key in sample.keys()])))
```

Comment:
Out of curiosity, is there an advantage to writing Python in this style?

If I'm understanding the code correctly, I would have written it like this myself:
```python
keys = set()
for sample in samples.values():
    for key in sample.keys():
        keys.update(key)
return sorted(list(keys))
```

Is this slower / less efficient or something? Or is it just down to personal taste?